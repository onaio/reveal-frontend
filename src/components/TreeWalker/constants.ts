export const ACTIVE = 'Active';
export const FEATURE = 'Feature';
export const FIND_BY_ID = 'location/findByJurisdictionIds';
export const FIND_BY_PROPERTIES = 'location/findByProperties';
export const LOCATION = 'location';
export const COULDNT_LOAD_CHILDREN = 'Could not load children';
export const COULDNT_LOAD_PARENTS = 'Could not load parents';
export const NOT_AVAILABLE = 'N/A';
