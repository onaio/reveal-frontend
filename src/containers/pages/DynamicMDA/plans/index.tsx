import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import {
  HIDDEN_PLAN_STATUSES,
  SUPERSET_DYNAMIC_MDA_REPORTING_PLANS_SLICE,
} from '../../../../configs/env';
import { MDA_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_MDA_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams, RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import GenericPLansReducer, {
  genericFetchPlans,
  makeGenericPlansArraySelector,
  reducerName as GenericPLansReducerName,
} from '../../../../store/ducks/generic/plans';
import { GenericPlan } from '../../../../store/ducks/generic/plans';
import { InterventionType } from '../../../../store/ducks/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the MDA plan definitions reducer */
reducerRegistry.register(GenericPLansReducerName, GenericPLansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);

/** selector for Dynamic MDA plans */
const MDAPlansArraySelector = makeGenericPlansArraySelector();

/** Simple component that loads a list of MDA plans */
const MDAPLansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for MDAPLansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: genericFetchPlans,
  pageTitle: MDA_PLANS,
  pageUrl: REPORT_MDA_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_DYNAMIC_MDA_REPORTING_PLANS_SLICE,
};

MDAPLansList.defaultProps = defaultProps;

export { MDAPLansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const plans = MDAPlansArraySelector(state, {
    interventionTypes: [InterventionType.DynamicMDA],
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: genericFetchPlans };

/** Connected MDA Plans component */
const ConnectedMDAPLansList = connect(mapStateToProps, mapDispatchToProps)(MDAPLansList);

export default ConnectedMDAPLansList;
