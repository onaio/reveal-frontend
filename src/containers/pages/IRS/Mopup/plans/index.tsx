import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { MOP_UP_REPORTING_TITLE } from '../../../../../configs/lang';
import { IRS_MOP_UP_REPORT_URL } from '../../../../../constants';
import { RouteParams } from '../../../../../helpers/utils';
import ConnectedIRSPlansList from '../../plans';

export const IRSMopUpReporting = (props: RouteComponentProps<RouteParams>) => {
  const plansProps = {
    ...props,
    pageTitle: MOP_UP_REPORTING_TITLE,
    pageUrl: IRS_MOP_UP_REPORT_URL,
  };
  return <ConnectedIRSPlansList {...plansProps} />;
};
