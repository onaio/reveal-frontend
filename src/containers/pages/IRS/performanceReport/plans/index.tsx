import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { IRS_PERFORMANCE_REPORTING_TITLE } from '../../../../../configs/lang';
import { PERFORMANCE_REPORT_IRS_PLAN_URL } from '../../../../../constants';
import { RouteParams } from '../../../../../helpers/utils';
import ConnectedIRSPlansList from '../../plans';

export const IRSPerfomenceReport = (props: RouteComponentProps<RouteParams>) => {
  const plansProps = {
    ...props,
    pageTitle: IRS_PERFORMANCE_REPORTING_TITLE,
    pageUrl: PERFORMANCE_REPORT_IRS_PLAN_URL,
  };
  return <ConnectedIRSPlansList {...plansProps} />;
};
