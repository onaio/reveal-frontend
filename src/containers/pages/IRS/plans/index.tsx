import ListView from '@onaio/list-view';
import reducerRegistry from '@onaio/redux-reducer-registry';
import React, { useEffect, useState } from 'react';
import { Helmet } from 'react-helmet';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Col, Row } from 'reactstrap';
import { Store } from 'redux';
import HeaderBreadcrumb from '../../../../components/page/HeaderBreadcrumb/HeaderBreadcrumb';
import Loading from '../../../../components/page/Loading';
import { SUPERSET_IRS_REPORTING_PLANS_SLICE } from '../../../../configs/env';
import { HOME, HOME_URL, IRS_PLANS, REPORT_IRS_PLAN_URL } from '../../../../constants';
import supersetFetch from '../../../../services/superset';
import IRSPlansReducer, {
  fetchIRSPlans,
  getIRSPlansArray,
  IRSPlan,
  reducerName as IRSPlansReducerName,
} from '../../../../store/ducks/generic/plans';

/** register the plan definitions reducer */
reducerRegistry.register(IRSPlansReducerName, IRSPlansReducer);

/** interface for PlanList props */
interface PlanListProps {
  fetchPlans: typeof fetchIRSPlans;
  plans: IRSPlan[];
  service: typeof supersetFetch;
}

/** Simple component that loads the new plan form and allows you to create a new plan */
const IRSPlansList = (props: PlanListProps) => {
  const { fetchPlans, plans, service } = props;
  const [loading, setLoading] = useState<boolean>(true);

  const pageTitle: string = IRS_PLANS;

  const breadcrumbProps = {
    currentPage: {
      label: pageTitle,
      url: REPORT_IRS_PLAN_URL,
    },
    pages: [
      {
        label: HOME,
        url: HOME_URL,
      },
    ],
  };

  /** async function to load the data */
  async function loadData() {
    try {
      setLoading(plans.length < 1); // only set loading when there are no plans
      await service(SUPERSET_IRS_REPORTING_PLANS_SLICE).then((result: IRSPlan[]) =>
        fetchPlans(result)
      );
    } catch (e) {
      // do something with the error?
    } finally {
      setLoading(false);
    }
  }

  useEffect(() => {
    loadData();
  }, []);

  if (loading === true) {
    return <Loading />;
  }

  const listViewProps = {
    data: plans.map(planObj => {
      return [
        <Link to={`${REPORT_IRS_PLAN_URL}/${planObj.plan_id}`} key={planObj.plan_id}>
          {planObj.plan_title}
        </Link>,
        planObj.plan_date,
        planObj.plan_effective_period_start,
        planObj.plan_effective_period_end,
        planObj.plan_status,
      ];
    }),
    headerItems: ['Title', 'Date Created', 'Start Date', 'End Date', 'Status'],
    tableClass: 'table table-bordered plans-list',
  };

  return (
    <div>
      <Helmet>
        <title>{pageTitle}</title>
      </Helmet>
      <HeaderBreadcrumb {...breadcrumbProps} />
      <Row>
        <Col md={8}>
          <h3 className="mt-3 mb-3 page-title">{pageTitle}</h3>
        </Col>
      </Row>
      <Row>
        <Col>
          <ListView {...listViewProps} />
        </Col>
      </Row>
    </div>
  );
};

/** Declare default props for IRSPlansList */
const defaultProps: PlanListProps = {
  fetchPlans: fetchIRSPlans,
  plans: [],
  service: supersetFetch,
};

IRSPlansList.defaultProps = defaultProps;

export { IRSPlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps {
  plans: IRSPlan[];
}

/** map state to props */
const mapStateToProps = (state: Partial<Store>): DispatchedStateProps => {
  const planDefinitionsArray = getIRSPlansArray(state);

  return {
    plans: planDefinitionsArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: fetchIRSPlans };

/** Connected ActiveFI component */
const ConnectedIRSPlansList = connect(
  mapStateToProps,
  mapDispatchToProps
)(IRSPlansList);

export default ConnectedIRSPlansList;
