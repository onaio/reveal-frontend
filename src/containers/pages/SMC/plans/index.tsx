import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import { HIDDEN_PLAN_STATUSES, SUPERSET_SMC_REPORTING_PLANS_SLICE } from '../../../../configs/env';
import { SMC_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_SMC_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams } from '../../../../helpers/utils';
import { RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import SMCPlansReducer, {
  fetchSMCPlans,
  makeSMCPlansArraySelector,
  reducerName as SMCPlansReducerName,
  SMCPLANType,
} from '../../../../store/ducks/generic/SMCPlans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the SMC plan definitions reducer */
reducerRegistry.register(SMCPlansReducerName, SMCPlansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);
// store.dispatch(fetchSMCPlans(testlans as any))

/** Simple component that loads a preview list of SMC plans */
const SMCPlansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for SMCPlansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: fetchSMCPlans,
  pageTitle: SMC_PLANS,
  pageUrl: REPORT_SMC_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_SMC_REPORTING_PLANS_SLICE,
};

SMCPlansList.defaultProps = defaultProps;

export { SMCPlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: SMCPLANType[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const SMCPlansArray = makeSMCPlansArraySelector()(state, {
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans: SMCPlansArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: fetchSMCPlans };

/** Connected ActiveFI component */
const ConnectedSMCPlansList = connect(mapStateToProps, mapDispatchToProps)(SMCPlansList);

export default ConnectedSMCPlansList;
