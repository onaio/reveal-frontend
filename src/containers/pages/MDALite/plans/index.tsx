import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import {
  HIDDEN_PLAN_STATUSES,
  SUPERSET_MDA_LITE_REPORTING_PLANS_SLICE,
} from '../../../../configs/env';
import { MDA_LITE_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_MDA_LITE_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams, RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import GenericPlansReducer, {
  genericFetchPlans,
  GenericPlan,
  makeGenericPlansArraySelector,
  reducerName as genericReducerName,
} from '../../../../store/ducks/generic/plans';
import { InterventionType } from '../../../../store/ducks/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the MDA plan definitions reducer */
reducerRegistry.register(genericReducerName, GenericPlansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);

/** selector for MDA lite plans */
const makeMDALitePlansArraySelector = makeGenericPlansArraySelector();

/** Simple component that loads a preview list of MDA plans */
const MDALitePlansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for MDALitePlansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: genericFetchPlans,
  pageTitle: MDA_LITE_PLANS,
  pageUrl: REPORT_MDA_LITE_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_MDA_LITE_REPORTING_PLANS_SLICE,
};

MDALitePlansList.defaultProps = defaultProps;

export { MDALitePlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const MDALitePlansArray = makeMDALitePlansArraySelector(state, {
    interventionTypes: [InterventionType.MDALite],
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans: MDALitePlansArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: genericFetchPlans };

/** Connected ActiveFI component */
const ConnectedMDALitePlansList = connect(mapStateToProps, mapDispatchToProps)(MDALitePlansList);

export default ConnectedMDALitePlansList;
