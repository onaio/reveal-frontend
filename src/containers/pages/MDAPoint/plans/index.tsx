import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import {
  HIDDEN_PLAN_STATUSES,
  SUPERSET_MDA_POINT_REPORTING_PLANS_SLICE,
} from '../../../../configs/env';
import { MDA_POINT_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_MDA_POINT_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams } from '../../../../helpers/utils';
import { RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import GenericPlansReducer, {
  genericFetchPlans,
  GenericPlan,
  makeGenericPlansArraySelector,
  reducerName as genericReducerName,
} from '../../../../store/ducks/generic/plans';
import { InterventionType } from '../../../../store/ducks/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the MDA plan definitions reducer */
reducerRegistry.register(genericReducerName, GenericPlansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);

/** selector for MDA point plans */
const makeMDAPointPlansArraySelector = makeGenericPlansArraySelector();

/** Simple component that loads a preview list of MDA plans */
const MDAPointPlansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for MDAPointPlansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: genericFetchPlans,
  pageTitle: MDA_POINT_PLANS,
  pageUrl: REPORT_MDA_POINT_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_MDA_POINT_REPORTING_PLANS_SLICE,
};

MDAPointPlansList.defaultProps = defaultProps;

export { MDAPointPlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const MDAPointPlansArray = makeMDAPointPlansArraySelector(state, {
    interventionTypes: [InterventionType.MDAPoint],
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans: MDAPointPlansArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: genericFetchPlans };

/** Connected ActiveFI component */
const ConnectedMDAPointPlansList = connect(mapStateToProps, mapDispatchToProps)(MDAPointPlansList);

export default ConnectedMDAPointPlansList;
