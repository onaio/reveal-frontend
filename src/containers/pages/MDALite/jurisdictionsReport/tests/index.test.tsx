import reducerRegistry from '@onaio/redux-reducer-registry';
import superset from '@onaio/superset-connector';
import { mount, shallow } from 'enzyme';
import flushPromises from 'flush-promises';
import { createBrowserHistory } from 'history';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import ConnectedMdaLiteJurisdictionReport, { MdaLiteJurisdictionReport } from '../';
import { REPORT_MDA_LITE_PLAN_URL } from '../../../../../constants';
import store from '../../../../../store';
import GenericJurisdictionsReducer, {
  fetchGenericJurisdictions,
  reducerName as GenericJurisdictionsReducerName,
} from '../../../../../store/ducks/generic/jurisdictions';
import { genericFetchPlans } from '../../../../../store/ducks/generic/plans';
import { GenericPlan } from '../../../../../store/ducks/generic/plans';
import { MDALitePlans } from '../../../../../store/ducks/superset/MDALite/tests/fixtures';
import { MDALiteJurReports } from './fixtures';

/* tslint:disable-next-line no-var-requires */
const fetch = require('jest-fetch-mock');

jest.mock('../../../../../configs/env');

const history = createBrowserHistory();

/** register the reducers */
reducerRegistry.register(GenericJurisdictionsReducerName, GenericJurisdictionsReducer);

store.dispatch(fetchGenericJurisdictions('1', MDALiteJurReports));
store.dispatch(genericFetchPlans(MDALitePlans as GenericPlan[]));

describe('components/MDA/Lite/Reports/JurisdictionReport', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', async () => {
    const mock: any = jest.fn();
    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: {
          planId: MDALitePlans[0].plan_id,
        },
        path: `${REPORT_MDA_LITE_PLAN_URL}/:planId`,
        url: `${REPORT_MDA_LITE_PLAN_URL}/${MDALitePlans[0].plan_id}`,
      },
    };
    shallow(
      <Router history={history}>
        <MdaLiteJurisdictionReport {...props} />
      </Router>
    );
  });

  it('renders correctly', async () => {
    fetch.mockResponseOnce(JSON.stringify({}));
    const mock: any = jest.fn();

    superset.api.doFetch = jest.fn(() => Promise.resolve({}));

    const supersetServiceMock: any = jest.fn();
    supersetServiceMock.mockImplementation(async () => []);
    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: {
          planId: MDALitePlans[0].plan_id,
        },
        path: `${REPORT_MDA_LITE_PLAN_URL}/:planId`,
        url: `${REPORT_MDA_LITE_PLAN_URL}/${MDALitePlans[0].plan_id}`,
      },
    };

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedMdaLiteJurisdictionReport {...props} />
        </Router>
      </Provider>
    );
    await act(async () => {
      await flushPromises();
    });
    wrapper.update();

    expect(JSON.stringify(wrapper.find('GenericJurisdictionReport').props())).toMatchSnapshot();
    wrapper
      .find('BreadcrumbItem li')
      .forEach((item, i) => expect(item.text()).toMatchSnapshot(`breadcrumb item-${i + 1}`));
    expect(wrapper.find('h3.page-title').text()).toEqual('MDA Lite Reporting: MDA-Lite 2021-03-02');
    expect(wrapper.find('.page-title').text()).toEqual('MDA Lite Reporting: MDA-Lite 2021-03-02');
    expect((wrapper.find('DrillDownTable').props().data as any).length).toEqual(4);

    expect(wrapper.find('DrillDownTable .thead .tr').length).toEqual(2);
    expect(wrapper.find('.thead .tr .th').length).toEqual(38);
    expect(wrapper.find('.no-data-cls').length).toMatchInlineSnapshot(`0`);
    // drilling down
    // country level
    expect(wrapper.find('.tbody .tr').length).toEqual(1);
    expect(wrapper.find('.tbody .tr .td').length).toEqual(21);
    expect(
      wrapper
        .find('.tbody .tr .td')
        .at(0)
        .text()
    ).toEqual('Kenya');
    wrapper
      .find('.tbody .tr .td')
      .at(0)
      .simulate('click');
    await act(async () => {
      await flushPromises();
    });
    wrapper.update();
    expect(wrapper.find('.page-title').text()).toEqual(
      'MDA Lite Reporting: MDA-Lite 2021-03-02: Kenya'
    );
    // county level
    expect(
      wrapper
        .find('.tbody .tr .td')
        .at(0)
        .text()
    ).toEqual('vihiga');
    wrapper
      .find('.tbody .tr .td')
      .at(0)
      .simulate('click');
    await act(async () => {
      await flushPromises();
    });
    wrapper.update();
    expect(wrapper.find('.page-title').text()).toEqual(
      'MDA Lite Reporting: MDA-Lite 2021-03-02: vihiga'
    );
    // sub-county level
    const rows = wrapper.find('.tbody .tr');
    expect(rows.length).toEqual(2);
    expect(
      rows
        .at(0)
        .find('.td')
        .at(0)
        .text()
    ).toEqual('emuhaya');
    expect(
      rows
        .at(1)
        .find('.td')
        .at(0)
        .text()
    ).toEqual('hamisi');
  });
});
