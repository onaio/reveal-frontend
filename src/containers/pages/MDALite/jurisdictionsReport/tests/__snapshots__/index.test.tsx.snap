// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MDA/Lite/Reports/JurisdictionReport renders correctly 1`] = `"{\\"history\\":{\\"length\\":1,\\"action\\":\\"POP\\",\\"location\\":{\\"pathname\\":\\"/\\",\\"search\\":\\"\\",\\"hash\\":\\"\\"}},\\"match\\":{\\"isExact\\":true,\\"params\\":{\\"planId\\":\\"5e396185-6094-4817-9dd4-24bcbbc698b0\\"},\\"path\\":\\"/intervention/mda-lite/report/:planId\\",\\"url\\":\\"/intervention/mda-lite/report/5e396185-6094-4817-9dd4-24bcbbc698b0\\"},\\"jurisdictions\\":[{\\"id\\":\\"165b0ae7-92fa-5b70-a7bb-283797f565c2\\",\\"plan_id\\":\\"5e396185-6094-4817-9dd4-24bcbbc698b0\\",\\"jurisdiction_id\\":\\"25e8a10b-737c-44e7-bfff-3b6344b1126f\\",\\"jurisdiction_parent_id\\":\\"\\",\\"jurisdiction_name\\":\\"Kenya\\",\\"jurisdiction_depth\\":0,\\"jurisdiction_path\\":[],\\"jurisdiction_name_path\\":[],\\"treated_male_1_4\\":\\"200\\",\\"treated_male_5_14\\":0,\\"treated_male_above_15\\":0,\\"treated_female_1_4\\":0,\\"treated_female_5_14\\":0,\\"treated_female_above_15\\":0,\\"total_males\\":20,\\"total_females\\":3,\\"total_all_genders\\":20,\\"supervisor_distributed\\":0,\\"received_number\\":0,\\"adminstered\\":0,\\"damaged\\":0,\\"adverse\\":0,\\"remaining_with_cdd\\":10,\\"returned_to_supervisor\\":3,\\"census_pop_target\\":200,\\"treatment_coverage\\":0.8,\\"other_pop_target\\":500,\\"other_pop_coverage\\":0.65},{\\"id\\":\\"89bb4010-38ee-5eac-b92a-4deafa62765a\\",\\"plan_id\\":\\"5e396185-6094-4817-9dd4-24bcbbc698b0\\",\\"jurisdiction_id\\":\\"38482d8c-e6ed-433b-860f-96c9cd09b4a3\\",\\"jurisdiction_parent_id\\":\\"3c63f48e-bb86-42e5-b3ff-bc539fbf5295\\",\\"jurisdiction_name\\":\\"emuhaya\\",\\"jurisdiction_depth\\":2,\\"jurisdiction_path\\":[\\"25e8a10b-737c-44e7-bfff-3b6344b1126f\\",\\"3c63f48e-bb86-42e5-b3ff-bc539fbf5295\\"],\\"jurisdiction_name_path\\":[\\"Kenya\\",\\"Vihiga\\"],\\"treated_male_1_4\\":0,\\"treated_male_5_14\\":0,\\"treated_male_above_15\\":0,\\"treated_female_1_4\\":0,\\"treated_female_5_14\\":0,\\"treated_female_above_15\\":0,\\"total_males\\":19,\\"total_females\\":100,\\"total_all_genders\\":0,\\"supervisor_distributed\\":0,\\"received_number\\":0,\\"adminstered\\":0,\\"damaged\\":0,\\"adverse\\":0,\\"remaining_with_cdd\\":0,\\"returned_to_supervisor\\":0,\\"census_pop_target\\":300,\\"treatment_coverage\\":0.9,\\"other_pop_target\\":400,\\"other_pop_coverage\\":0.6},{\\"id\\":\\"541e7ed0-2dc6-586e-b563-4c988c5ac585\\",\\"plan_id\\":\\"5e396185-6094-4817-9dd4-24bcbbc698b0\\",\\"jurisdiction_id\\":\\"3c63f48e-bb86-42e5-b3ff-bc539fbf5295\\",\\"jurisdiction_parent_id\\":\\"25e8a10b-737c-44e7-bfff-3b6344b1126f\\",\\"jurisdiction_name\\":\\"vihiga\\",\\"jurisdiction_depth\\":1,\\"jurisdiction_path\\":[\\"25e8a10b-737c-44e7-bfff-3b6344b1126f\\"],\\"jurisdiction_name_path\\":[\\"Kenya\\"],\\"treated_male_1_4\\":0,\\"treated_male_5_14\\":0,\\"treated_male_above_15\\":0,\\"treated_female_1_4\\":0,\\"treated_female_5_14\\":0,\\"treated_female_above_15\\":0,\\"total_males\\":0,\\"total_females\\":0,\\"total_all_genders\\":0,\\"supervisor_distributed\\":0,\\"received_number\\":0,\\"adminstered\\":0,\\"damaged\\":0,\\"adverse\\":0,\\"remaining_with_cdd\\":0,\\"returned_to_supervisor\\":0},{\\"id\\":\\"c477af78-93ce-51f8-bf1e-0bae3a80fcae\\",\\"plan_id\\":\\"5e396185-6094-4817-9dd4-24bcbbc698b0\\",\\"jurisdiction_id\\":\\"f2482d8c-e6ed-433b-860f-96c9cd09b4a3\\",\\"jurisdiction_parent_id\\":\\"3c63f48e-bb86-42e5-b3ff-bc539fbf5295\\",\\"jurisdiction_name\\":\\"hamisi\\",\\"jurisdiction_depth\\":2,\\"jurisdiction_path\\":[\\"25e8a10b-737c-44e7-bfff-3b6344b1126f\\",\\"3c63f48e-bb86-42e5-b3ff-bc539fbf5295\\"],\\"jurisdiction_name_path\\":[\\"Kenya\\",\\"Vihiga\\"],\\"treated_male_1_4\\":0,\\"treated_male_5_14\\":0,\\"treated_male_above_15\\":0,\\"treated_female_1_4\\":0,\\"treated_female_5_14\\":0,\\"treated_female_above_15\\":0,\\"total_males\\":0,\\"total_females\\":0,\\"total_all_genders\\":0,\\"supervisor_distributed\\":0,\\"received_number\\":0,\\"adminstered\\":0,\\"damaged\\":0,\\"adverse\\":0,\\"remaining_with_cdd\\":0,\\"returned_to_supervisor\\":0}],\\"plan\\":{\\"plan_id\\":\\"5e396185-6094-4817-9dd4-24bcbbc698b0\\",\\"jurisdiction_root_parent_ids\\":\\"[\\\\\\"bac41d87-b09d-4e76-be3b-3b6d07399891\\\\\\"]\\",\\"plan_title\\":\\"MDA-Lite 2021-03-02\\",\\"plan_name\\":\\"MDA-Lite-2021-03-02\\",\\"plan_status\\":\\"active\\",\\"plan_date\\":1614643200000,\\"plan_effective_period_start\\":1614643200000,\\"plan_effective_period_end\\":1616371200000,\\"plan_intervention_type\\":\\"MDA-Lite\\"},\\"baseURL\\":\\"/intervention/mda-lite/report\\",\\"focusAreaColumn\\":\\"mdaLitewardColumns\\",\\"jurisdictionColumn\\":\\"mdaLiteJurisdictionsColumns\\",\\"pageTitle\\":\\"MDA Lite Reporting\\",\\"reportingPlanSlice\\":\\"1\\",\\"slices\\":[\\"1\\",\\"2\\"]}"`;

exports[`components/MDA/Lite/Reports/JurisdictionReport renders correctly: breadcrumb item-1 1`] = `"Home"`;

exports[`components/MDA/Lite/Reports/JurisdictionReport renders correctly: breadcrumb item-2 1`] = `"MDA Lite Reporting"`;

exports[`components/MDA/Lite/Reports/JurisdictionReport renders correctly: breadcrumb item-3 1`] = `"MDA-Lite 2021-03-02"`;
